tests/test_repo_legit.py

```diff
@@ -0,0 +1,126 @@
+from os.path import expanduser, isdir, exists, join, abspath, dirname
+from os import makedirs, chmod, walk
+from git import Repo
+import sys
+import legit
+from shutil import rmtree
+import unittest
+
+
+url = 'https://github.com/katridi/test_folder'
+folder = join(expanduser('~'), 'Desktop\\folder2')
+path = abspath(dirname(dirname(abspath(__file__))))
+sys.path.append(path)
+
+def create_folder(folder: str) -> str:
+    if not exists(folder):
+        makedirs(folder)
+    assert isdir(folder)
+
+def change_mod_recursively(folder: str, permissions: oct = 0o777) -> None:
+    for root, dirs, files in walk(folder):
+        for d in dirs:
+            chmod(join(root, d), permissions)
+        for f in files:
+            chmod(join(root, f), permissions)
+
+
+class TestGitMethods(unittest.TestCase):
+
```

Comment:
You'd probably want a setup method that creates a dirs for origin repo and working repo. You don't need to test creation of dirs in your tests for repo operations.